Library/Homebrew/livecheck/strategy.rb

```diff
@@ -180,22 +175,17 @@ def self.page_headers(url, homebrew_curl: false)
         headers = []
 
         [:default, :browser].each do |user_agent|
-          stdout, _, status = curl_with_workarounds(
+          output, _, status = curl_with_workarounds(
             *PAGE_HEADERS_CURL_ARGS, url,
             **DEFAULT_CURL_OPTIONS,
             use_homebrew_curl: homebrew_curl,
             user_agent:        user_agent
           )
+          next unless status.success?
 
-          while stdout.match?(/\AHTTP.*\r$/)
-            h, stdout = stdout.split("\r\n\r\n", 2)
-
-            headers << h.split("\r\n").drop(1)
-                        .to_h { |header| header.split(/:\s*/, 2) }
-                        .transform_keys(&:downcase)
-          end
-
-          return headers if status.success?
+          parsed_output = parse_curl_output(output)
+          parsed_output[:heads].each { |head| headers << head[:headers] }
+          return headers if headers.present?
```

Comment:
```suggestion
          headers = parsed_output[:heads].map { |head| head[:headers] }
```

Reply:
No need to `return` when you're doing so immediately below.

Reply:
The `return` in the loop ensures that we only fetch the URL using a different user-agent (i.e., `:browser`) if the fetch using the `:default` user-agent doesn't work properly. The suggested change here would cause all `#page_headers` requests to be fetched twice (once with each user agent), even if the request works fine with the `:default` user-agent.

That said, the fallback user-agent code here (and in `#page_content`) will be removed after I've added the ability to specify the user-agent in a `livecheck` block. I've had this implemented locally for a while but I have a series of brew PRs in the pipeline to work through beforehand and I want to get this `curl` PR wrapped up before I start on that.

Reply:
Ok, can leave it if this is going to go away.